### DISCUSSION TITLE
C# Markup Reference? Also Textbox

### DISCUSSION THREAD

**Discussion by JohnWimbish** [unoplatform/uno#14823] [Q&A] [answered]
Is there a document describing in detail C# equivalents for XAML? Much of the documentation is XAML only.

E.g., here: "https://platform.uno/docs/articles/external/uno.themes/doc/material-controls-extensions.html" gives xaml for placing an Icon in a TextBox, but I can't figure out how to do it in C#.

**Comment by francoistanguay** [comment, marked as answer]
C# Markup sorrce generators automaticcaly generate one extension method per class containing attached properties. So you should have a .ControlExtensions(icon = ...) available to you.

FOr info here: https://platform.uno/docs/articles/external/uno.extensions/doc/Overview/Markup/AttachedProperties.html

You'll need to make sure you have the Markup version of the Toolkit package referenced where ControlExtensions is defined.

cc @dansiegel  @kazo0

**Reply by JohnWimbish** [reply]
Thanks....That did the trick. 
```
  new TextBox()
       .Text("Hello")
       .ControlExtensions(icon: new SymbolIcon(Symbol.SolidStar))
```

**Reply by francoistanguay** [reply]
Amazing! As you encounter new issues/questions and go through our docs, feel free to let us know of any ways we can imrpvoe for ease of understanding!
